dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java

```diff
@@ -115,13 +130,57 @@ public class DeviceServiceImpl extends AbstractCachedEntityService<DeviceCacheKe
     @Autowired
     private EntityCountService countService;
 
+    @Autowired
+    private DeviceConnectivityConfiguration deviceConnectivityConfiguration;
+
     @Override
     public DeviceInfo findDeviceInfoById(TenantId tenantId, DeviceId deviceId) {
         log.trace("Executing findDeviceInfoById [{}]", deviceId);
         validateId(deviceId, INCORRECT_DEVICE_ID + deviceId);
         return deviceDao.findDeviceInfoById(tenantId, deviceId.getId());
     }
 
+    @Override
+    public Map<String, String> findDevicePublishTelemetryCommands(String baseUrl, Device device) throws URISyntaxException {
```

Comment:
Consider to move this functionality to a separate service. DeviceServiceImpl is already pretty big